### ISSUE TITLE
unknown tag <!secret>YAML

### ISSUE THREAD

**Issue by Salvora** [esphome/esphome-vscode#60]
This issue has been reported before in https://github.com/esphome/esphome-vscode/issues/16 but the suggested fix did not work for me.

I am getting these syntax errors for the "!secret tag".

My yaml custom tags:

```yaml
    "yaml.customTags": [
      "!env_var scalar",
      "!include_dir_list scalar",
      "!include_dir_merge_list scalar",
      "!include_dir_merge_named scalar",
      "!include_dir_named scalar",
      "!include scalar",
      "!include_dir_merge_named",
      "!secret scalar"
    ],
```

The exact error:
```log
Secret 'esp_wifi_ssid' not defined
```

![image](https://user-images.githubusercontent.com/6360098/194731452-125b09ac-c84c-4ac9-8db0-d14eea0f3d35.png)

**Comment by glmnet** [comment]
Btw this is not a syntax error not neither a secrets unknown tag error!! Please fix title and try to understand the message properly

**Comment by Salvora** [comment]
Changed the title, dunno if this suits better.
Add-on config:
```yaml
ssl: true
certfile: fullchain.pem
keyfile: privkey.pem
leave_front_door_open: true
status_use_ping: false
```

**Comment by glmnet** [comment]
that's ok, so you are running the ESPHome add on inside a HA docker or Home Assistant OS.

I'm wondering actually how are you configuring the ESPHome VS code extension, is validator local or dashboard?

where is physically (os path) the file you are validating and where is physically the secrets.yaml file? (they should be in the same folder)

**Comment by Salvora** [comment]
I use Home Assistant OS
ESPHome VS code extension is configured as dashboard.
It is accessible with a valid SSL certificate through nginx.
Extension successfully connects to the dashboard.
I tested by converting all the secrets to strings and everything seems to work then.

**Comment by glmnet** [comment]
What you mean that converted to string? Is your ssid or password a number?

**Comment by Salvora** [comment]
I mean instead of using !secret tag I used the actual passwords in plain text.

**Comment by glmnet** [comment]
But your secrets work or not? This seems like ESPHome cannot read your secrets at all, I mean not a vscode only issue but you have something wrong with your setup or secrets file contents 

To put it clear: Are you able to compile and upload if you just ignore the errors shown in vs code?

**Comment by Salvora** [comment]
Yes, I can compile and upload just fine and the secrets work just fine. It is just a visual error on the vscode side.

**Comment by Salvora** [comment]
The same thing is also happening with the vscode addon for the home assistant. If I open an esphome configuration yaml file and switch the language mode from yaml to esphome to view those files. I get the same errors.

**Comment by glmnet** [comment]
I don't understand what's going on then. 

When vscode validates your yaml it will send all the file to ESPHome inside the dashboard, then it will realize you have secrets in your yaml and will try to open the secrets.yaml locally (I.e. inside the dashboard itself). Then it will read the secret exists and value is ok. All this happens inside the ESPHome dashboard, then if there is an error it will send back this error information to the vscode extension and it will just print the error there. The vscode extension does not do any validation by itself, so it's weird you get this error while validating but then compilation works.

**Comment by Salvora** [comment]
Can you replicate this on your config?

**Comment by glmnet** [comment]
You're right. I'm confused there are three kinds of validations that I've worked on and mixed up my head

I tried to reproduce your scenario but could not get the errors you've got

Can you give more details on your environment? 

Are you using studio code add on or is vs code running on a pc / Mac? How files are shared? Which folder do you open in code?

Thanks

**Comment by glmnet** [comment]
You're right, I finally reproduced this

**Comment by Salvora** [comment]
Vscode windows, SMB share, config folder
